### PR TITLE
random: add seed() for deterministic PRNG

### PR DESCRIPTION
**Pull request by matslina**
This makes the RNG thread-local, and by default reseeded on each applet execution.
User has the option of setting their own seed, which lasts for 1 execution of the applet.

### CODE REVIEW

File: runtime/modules/random/random_test.go [hunk 1]
```diff
@@ -4,24 +4,43 @@ import (
...
+def test_seed():
+    random.seed(4711)
+    sequence = [random.number(0, 1 << 20) for _ in range(500)]
+
+    random.seed(4711) # same seed
+    for i in range(len(sequence)):
+        if sequence[i] != random.number(0, 1 << 20):
+            fail("sequence mismatch despite identical seed")
+
+    random.seed(4712) # diferent seed
```
Comment (matslina) [inline runtime/modules/random/random_test.go:34]:
```suggestion
    random.seed(4712) # different seed
```

**Review by betterengineering — APPROVED** [review]
🙌

File: runtime/modules/random/random.go [hunk 2]
```diff
@@ -70,7 +107,10 @@ func randomNumber(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tu
 		return nil, fmt.Errorf("max is less then min")
 	}
 
-	number := rand.Int63n(max-min+1) + min
+	rng, ok := thread.Local(threadRandKey).(*rand.Rand)
+	if !ok || rng == nil {
+		return nil, fmt.Errorf("RNG not set (very bad!)")
```
Comment (rohansingh) [inline runtime/modules/random/random.go:112]: image

File: runtime/modules/random/random.go [hunk 3]
```diff
@@ -11,14 +11,24 @@ import (
...
+	ModuleName    = "random"
+	threadRandKey = "tidbyt.dev/pixlet/runtime/random"
 )
 
 var (
 	once   sync.Once
 	module starlark.StringDict
 )
 
+func AttachRandToThread(t *starlark.Thread) {
```
Comment (rohansingh) [inline runtime/modules/random/random.go:23]:
```suggestion
func AttachToThread(t *starlark.Thread) {
```

nit

File: runtime/applet.go [hunk 4]
```diff
@@ -77,6 +77,8 @@ func (a *Applet) thread(initializers ...ThreadInitializer) *starlark.Thread {
 		a.decrypter.attachToThread(t)
 	}
 
+	random.AttachRandToThread(t)
```
Comment (matslina) [inline runtime/applet.go:80]:
```suggestion
	random.AttachToThread(t)
```

**Pull request merged**